src/sentry/incidents/subscription_processor.py

```diff
@@ -716,11 +742,13 @@ def handle_trigger_actions(
             if incident_trigger.status == TriggerStatus.ACTIVE.value
             else AlertRuleTriggerActionMethod.RESOLVE.value
         )
-        try:
+
+        # NOTE: all incident_triggers are derived from self.active_incident
+        incident = self.active_incident
```

Comment:
Are you sure there's not some other reason we do this? Is there's no active incident at the moment does it cause a problem?

Reply:
Ah yea - so previously the `trigger_resolve_threshold` would clear teh `active_incident` before we got to this point
This is why we fetched the incident previously.

BUT - the incident fetched via the `incident_trigger` here _is just the active incident_ 
So - rather than making another query to grab the `Incident`, i figured we could just grab the available active incident.

Reply:
Could we explain this more explicitly in the comments? 

"We attempt to fetch the active incident here to save a query. When this is triggered by a resolve, there will be no incident so we fall back to fetching from the db"

Reply:
I'm also wondering if we explicitly want to reload it from the db here though...

Reply:
still not sure reloading it is necessary since it's explicitly being set and utilized in the previous steps (`trigger_alert_threshold`)